app/src/main/java/io/zenandroid/onlinego/ui/screens/stats/StatsFragment.kt

```diff
@@ -34,5 +73,301 @@ class StatsFragment : Fragment(), StatsContract.View {
         presenter.unsubscribe()
     }
 
+    override fun fillPlayerDetails(playerDetails: OGSPlayer ) {
+        nameView.text = playerDetails.username
+        rankView.text = formatRank(egfToRank(playerDetails.ratings?.overall?.rating))
+        playerDetails.country?.let {
+            flagView.text = convertCountryCodeToEmojiFlag(it)
+        }
+        playerDetails.icon?.let {
+            Glide.with(this)
+                    .load(processGravatarURL(it, iconView.width))
+                    .transition(DrawableTransitionOptions.withCrossFade(DrawableCrossFadeFactory.Builder().setCrossFadeEnabled(true).build()))
+                    .apply(RequestOptions().centerCrop().placeholder(R.drawable.ic_person_outline))
+                    .into(iconView)
+        }
+
+        var rankViewText = SpannableString("${formatRank(egfToRank(playerDetails.ratings?.overall?.rating))}\n" +
+                "${playerDetails.ratings?.overall?.rating?.toInt()} ± ${playerDetails.ratings?.overall?.deviation?.toInt()}")
+        rankViewText.apply {
+            listOf(AbsoluteSizeSpan(20, true), StyleSpan(BOLD)).forEach { span ->
+                setSpan(span,0, rankViewText.indexOf('\n'), SPAN_EXCLUSIVE_EXCLUSIVE)
+            }
+        }
+        rankViewText.apply {
+            listOf(AbsoluteSizeSpan(12, true), StyleSpan(NORMAL), ForegroundColorSpan(resources.getColor(R.color.colorTextSecondary))).forEach { span ->
+                setSpan(span, rankViewText.indexOf('\n'), rankViewText.length, SPAN_EXCLUSIVE_EXCLUSIVE)
+            }
+        }
+        rankView.text = rankViewText
+    }
+
+    override fun fillPlayerStats(playerStats: String): Pair<Long, Long> {
+        var gameList = playerStats.split('\n').drop(1).dropLast(2) // drops header,initialRating,trailing endline
```

Comment:
This shouldn't be a concern for the fragment, this needs to be handled somewhere in the retrofit or at most the service layer.

Reply:
Fixed in https://github.com/acristescu/OnlineGo/commit/1c57f6a852393483b9a7b3936e3048083c23aed1